contracts/RewardManager.sol

```diff
@@ -0,0 +1,273 @@
+// SPDX-License-Identifier: MIT
+pragma solidity = 0.7.6;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+
+contract RewardManager is Ownable, ReentrancyGuard
+{
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    // whitelisted rewardDistributors
+    mapping (address => bool) public rewardDistributor;
+    
+    // Call from excludedAddresses will be whitelisted & rewards harvested from farm will not be vested
+    mapping (address => bool) public excludedAddresses;
+    
+    // preMaturePenalty will be sent to burner address
+    address public l2Burner;
+    
+    //Upfront rewards unlock in percentage. This number is later divided by 1000 for calculations.
+    uint256 public upfrontUnlock;
+    
+    //Pre mature penalty in percentage. This number is later divided by 1000 for calculations.
+    uint256 public preMaturePenalty;
+
+    //Bonus Rewards in percentage. This number is later divided by 1000 for calculations.
+    uint256 public bonusPercentage;
+    
+    /// @notice start of Distribution phase as a timestamp
+    uint256 public startDistribution;
+
+    /// @notice end of Distribution phase as a timestamp
+    uint256 public endDistribution;
+    
+    //Cryption Network Token (cnt) token address
+    IERC20 public cnt;
+    
+    /// @notice amount vested for a user.
+    mapping(address => uint256) public vestedAmount;
+    
+    /// @notice cumulative total of tokens drawn down (and transferred from the deposit account) per beneficiary
+    mapping(address => uint256) public totalDrawn;
+
+    /// @notice total tokens burnt per beneficiary
+    mapping(address => uint256) public burntAmount;
+
+    /// @notice bonus rewards entitled per beneficiary
+    mapping(address => uint256) public bonusReward;
+
+    /// @notice event emitted when a vesting schedule is created
+    event Vested(address indexed _beneficiary, uint256 indexed value);
+    
+    /// @notice event emitted when a successful drawn down of vesting tokens is made
+    event DrawDown(address indexed _beneficiary, uint256 indexed _amount, uint256 indexed bonus);
+    
+     /// @notice event emitted when a successful pre mature drawn down of vesting tokens is made
+    event PreMatureDrawn(address indexed _beneficiary, uint256 indexed burntAmount, uint256 indexed userEffectiveWithdrawn);
+
+    modifier checkPercentages(uint256 _upfrontUnlock, uint256 _preMaturePenalty) {
+        require(_upfrontUnlock.add(_preMaturePenalty) <= 1000, "Invalid Percentages");
```

Comment:
Is this condition valid? 

`upFrontUnlock` signifies the percentage the amount which is not vested. `preMaturePenalty` is percentage of "vested" amount which will be burned. Two are different things.

Reply:
yeah so , placed this modifier to always check whenever updating any Percentages the combination doesn't exceeds 100 %